android/app/src/main/java/app/organicmaps/car/screens/download/DownloaderHelpers.java

```diff
@@ -16,14 +17,9 @@ public final class DownloaderHelpers
 {
   static final String[] WORLD_MAPS = new String[]{"World", "WorldCoasts"};
 
-  // World maps may be missing only in the F-Droid build.
-  @SuppressWarnings("ConstantConditions")
   public static boolean isWorldMapsDownloadNeeded()
   {
-    // TODO: Maps are asynchronously initialized in the core. If the initialization takes a significant amount of time, the downloader screen could potentially be displayed, even if the world maps are present.
-    if (BuildConfig.FLAVOR.equals("fdroid"))
-      return !CountryItem.fill(WORLD_MAPS[0]).present || !CountryItem.fill(WORLD_MAPS[1]).present;
-    return false;
+    return DownloadResourcesLegacyActivity.nativeGetBytesToDownload() != 0;
```

Comment:
Looks a bit scary. If we need to check World files on disk, better to do it manually.
DownloadResourcesLegacyActivity native implementation is very specific and for example operates with a bunch of static vars, which can cause strange behavior later in the main app ..

Reply:
@vng can you suggest a better C++ way to check for world presence?

Reply:
Actually, the initial idea was to use a smart wrapper like nativeGetBytesToDownload, to check if all necessary resources are there (it could be not only world files, but also, for example, some fonts, or other data types). It also properly handles some previous user actions.

@Jean-BaptisteC can you please test this change as the fdroid build and as a google/web build?